bundles/org.eclipse.cdt.lsp.editor.ui/src/org/eclipse/cdt/lsp/editor/ui/clangd/ClangdConfigurationManager.java

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Bachmann electronic GmbH and others.
+ * 
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ * Gesa Hentschke (Bachmann electronic GmbH) - initial implementation
+ *******************************************************************************/
+
+package org.eclipse.cdt.lsp.editor.ui.clangd;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.PrintWriter;
+import java.util.Map;
+import java.io.IOException;
+
+import org.yaml.snakeyaml.Yaml;
+
+public class ClangdConfigurationManager {
+	public static final String CLANGD_CONFIG_FILE_NAME = ".clangd";
+	private static final String COMPILE_FLAGS = "CompileFlags";
+	private static final String COMPILATTION_DATABASE = "CompilationDatabase";
+	private static final String SET_COMPILATION_DB = COMPILE_FLAGS + ": {" + COMPILATTION_DATABASE + ": %s}";
+	
+	/**
+	 * Set the <code>CompilationDatabase</code> entry in the .clangd file in the given configFileDirectory 
+	 * to databasePath. The file will be created, if it's not existing. 
+	 * A ScannerException will be thrown if the configuration file contains invalid yaml syntax.
+	 * 
+	 * @param configFileDirectory
+	 * @param databasePath
+	 * @throws IOException, ScannerException
+	 */
+	@SuppressWarnings("unchecked")
+	public static void setCompilationDatabase(String configFileDirectory, String databasePath) throws IOException {
+		var configFile = new File(configFileDirectory, CLANGD_CONFIG_FILE_NAME);
```

Comment:
A refresh of configFile at the end of this method is needed (or write to an IFile instead of a java.io.File)

Reply:
You're right. I'll use an IFile instead.